learn/Understanding-Oracles-In-Web3-By-Paul-Elisha.md

```diff
@@ -0,0 +1,101 @@
+## Understanding Oracles in Web3: A Comprehensive Guide to Pull and Push-Based Oracles on the Kaia Chain
+
+## Introduction to Oracles
+
+In the evolving landscape of blockchain technology, oracles serve as essential intermediaries that enable smart contracts to interact with external data sources. They bridge the gap between blockchain networks and real-world information, allowing decentralized applications (dApps) to access off-chain data. This capability is crucial for executing smart contracts based on real-world events, such as price fluctuations, weather conditions, or sports results.
+
+Oracles can be classified into various types based on their functionality, including pull-based and push-based oracles. Each type has its own advantages and use cases, making them integral to the functionality of decentralized ecosystems like the Kaia Chain.
+
+## The Role of Oracles in Web3
+
+Oracles act as a conduit for data, enabling smart contracts to respond dynamically to external conditions. Without oracles, smart contracts would be limited to operating solely on the data available within their blockchain environment. This limitation would severely restrict their applicability in real-world scenarios.
+
+### Key Functions of Oracles
+`**Data Retrieval:**` Oracles gather data from various external sources, including APIs, databases, and sensors.
+`**Data Verification:**` They validate the authenticity of the data before relaying it to the blockchain.
+`**Execution Triggering:**` Oracles trigger smart contract execution based on predefined conditions linked to the incoming data.
+
+## Types of Oracles
+
+Oracles can be categorized based on several criteria:
+
+### Inbound vs. Outbound:
+
+`**Inbound Oracles:**` These oracles transmit external data into the blockchain.
+
+`**Outbound Oracles:**` They send information from the blockchain to external systems.
+
+### Centralized vs. Decentralized:
+
+`**Centralized Oracles:**` Controlled by a single entity, posing risks related to trust and reliability.
+
+`**Decentralized Oracles:**` Utilize multiple sources for data collection, enhancing security and reducing single points of failure.
+
+### Push vs. Pull:
+
+`**Push-Based Oracles:**` Proactively send updates to smart contracts without explicit requests.
+
+`**Pull-Based Oracles:**` Require smart contracts to request data explicitly.
+
+### Push-Based Oracles
```

Comment:
can you give an example of oracle providers using `Push-Based Oracles` in this category? Same thing goes for Pull-Based Oracles